core/src/frame_lifecycle.rs

```diff
@@ -85,6 +87,8 @@ pub fn run_all_phases_avm2(context: &mut UpdateContext<'_, '_>) {
 
     stage.enter_frame(context);
 
+    let mut ran_framescript = HashSet::new();
```

Comment:
Would `FnvHashSet` make more sense here? I think we generally use FnvHash over Hash when we prefer speed over potential DOS attacks

Reply:
Good point - I've changed it